src/modules/navigator/navigation.h

```diff
@@ -142,11 +143,20 @@ struct mission_item_s {
 				float pitch_min;		/**< minimal pitch angle for fixed wing takeoff waypoints */
 				float circle_radius;		/**< geofence circle radius in meters (only used for NAV_CMD_NAV_FENCE_CIRCLE*) */
 			};
-			float acceptance_radius;	/**< default radius in which the mission is accepted as reached in meters */
-			float loiter_radius;		/**< loiter radius in meters, 0 for a VTOL to hover, negative for counter-clockwise */
 			float yaw;					/**< in radians NED -PI..+PI, NAN means don't change yaw		*/
-			float ___lat_float;			/**< padding */
-			float ___lon_float;			/**< padding */
+			float acceptance_radius;	/**< default radius in which the mission is accepted as reached in meters */
+			union {
+				struct {
+					float loiter_radius;		/**< loiter radius in meters, 0 for a VTOL to hover, negative for counter-clockwise */
+					float ___lat_float;			/**< padding */
+					float ___lon_float;			/**< padding */
+				};
+				struct {
```

Comment:
I need to dig into AMS a bit more, but you likely don't need to extend `mission_item_s` like this unless you're checking for velocity setup acceptance (`MissionBlock::is_mission_item_reached()`).

Reply:
How should I do then? I wanted to add vx, vy and vz fields without increasing the size of mission_item_s. That's why I selected three floats I don't need for the velocity waypoint to put them in the same union.